src/main/java/ui/components/FilterTextField.java

```diff
@@ -94,6 +95,20 @@ private void setup() {
     }
 
     /**
+     * Set the background of FilterTextField to white if there is no syntax errors.
+     * Otherwise, background is set to red
+     *
+     * @param isValidFilter
+     */
+    public void setTextFieldBackground(boolean isValidFilter){
+        if (isValidFilter) {
+            this.setStyle("-fx-control-inner-background: white");
+        } else {
+            this.setStyle("-fx-control-inner-background: #EE8993");
```

Comment:
setTextFieldBackground is also used in FilterPanel to change the background colour of the field text after the user presses enter.